src/main/java/io/fabric8/elasticsearch/plugin/filter/FieldStatsResponseFilter.java

```diff
@@ -72,6 +73,17 @@ public void apply(final Task task, final String action, final ActionRequest requ
             @SuppressWarnings("unchecked")
             @Override
             public void onResponse(ActionResponse response) {
+                if(INDICES_FIELD_STATS_READ_ACTION.equals(action) && response instanceof FieldStatsResponse) {
```

Comment:
There is a refactoring opportunity with this and onFailure since we appear to be checking the same thing